receptorctl/receptorctl/socket_interface.py

```diff
@@ -9,10 +9,22 @@
 import yaml
 
 class ReceptorControl:
-    def __init__(self):
+    def __init__(self, socketaddress, config=None, tlsclient=None, rootcas=None, key=None, cert=None, insecureskipverify=False):
+        if config and any((rootcas, key, cert)):
+            raise RuntimeError("Cannot specify both config and rootcas, key, cert")
+        if config and not tlsclient:
+            raise RuntimeError("Must specify both config and tlsclient")
         self.socket = None
         self.sockfile = None
         self.remote_node = None
+        self.socketaddress = socketaddress
+        self.rootcas = rootcas
+        self.key = key
+        self.cert = cert
+        self.insecureskipverify = insecureskipverify
+        if config and tlsclient:
+            self.readconfig(config, tlsclient)
+        self.connect()
```

Comment:
is it okay to go ahead and call connect here?

Reply:
I think it would be better to track connection state and lazily connect when someone calls a method that requires a connection.

Reply:
Updated to remove `connect` on init. 
The way I'm tracking connection state is by checking if `self.socket` is None in the connect method. Do you think that suffices?

Reply:
Yes, I think so, but see my other comments.  We need to make sure to set `self.socket` back to None in cases where the socket remains open, but is no longer usable for subsequent commands.